state/pool.go

```diff
@@ -365,7 +377,8 @@ func (p *StatePool) Remove(modelUUID string) (bool, error) {
 	if !ok {
 		// Don't require the client to keep track of what we've seen -
 		// ignore unknown model uuids.
-		return false, nil
+		_, ok := p.tombstones[modelUUID]
+		return ok, nil
 	}
 	if !item.remove {
 		item.remove = true
```

Comment:
We don't need this member at all now that we always remove it when we have zero references.

Reply:
We need it for the removing channel. I'll swap it for a sync.Once

Reply:
So there is nuance here, we do use it in multiple locations as a side-effect from the removal call. I'm don't think we should spend too much effort on removing this one field though.